numpy/ma/extras.py

```diff
@@ -736,7 +736,7 @@ def _median(a, axis=None, out=None, overwrite_input=False):
     # duplicate high if odd number of elements so mean does nothing
     odd = counts % 2 == 1
     if asorted.ndim > 1:
-        np.copyto(low, high, where=odd)
+        low[odd] = high[odd]
```

Comment:
I am afraid I haven't got a grasp of the issue as good as yours, but what about replacing `h - 1` by `np.maximum(h - 1, 0)` ?